spec/frodo/concerns/api_spec.rb

```diff
@@ -116,67 +117,59 @@
     let(:attributes) { {} }
     let(:options) { attributes }
     let(:headers) { { 'odata-entityid' => url } }
-    let(:entity_set_name) { 'things' }
 
+    before{stub_request(verb, "#{connection_uri}/#{entity_type}").to_return(body: body.to_json, headers: headers)}
     subject { client.create!(entity_type, attributes) }
 
-    before do
-      allow(client).to receive(:service).and_return(service)
-      allow(service).to receive(:[]).with(entity_type).and_return(entity_set)
-      allow(entity_set).to receive(:name).and_return(entity_set_name)
-      allow(client).to receive(:entity_set_to_url_chunk).with(entity_set).and_return(path)
-    end
-
     it 'posts entity_set info and returns resulting id' do
       expect(subject).to eq(id)
     end
 
-    it 'raises errors that occur' do
-      allow(client).to receive(:service).and_raise(StandardError)
-
-      expect{ subject }.to raise_error(StandardError)
-    end
-
     context 'alias .insert!' do
+
       subject { client.insert!(entity_type, attributes) }
 
       it { should eq(id) }
     end
   end
 
   describe '.update' do
-    let(:attributes) { {} }
+    let(:attributes) { {'typeid' => 'some_id'} }
+    let(:primary_key) { 'typeid' }
 
-    subject { client.update(entity_type, attributes) }
+    before { stub_request(verb, "#{connection_uri}/#{entity_type}(some_id)").to_return(body: body.to_json, headers: headers) }
+    subject { client.update(entity_type, primary_key, attributes) }
 
     it 'calls .create! and returns the result' do
-      allow(client).to receive(:update!).with(entity_type, attributes).and_return(true)
+      expect(client).to receive(:update!).with(entity_type, primary_key, attributes).and_return(true)
 
       expect(subject).to be(true)
     end
 
     it 'returns false for expected exceptions' do
-      allow(client).to receive(:update!).with(entity_type, attributes).and_raise(client_error)
+      allow(client).to receive(:update!).with(entity_type, primary_key, attributes).and_raise(client_error)
 
       expect(subject).to eq(false)
     end
 
     it 'raises other errors' do
-      allow(client).to receive(:update!).with(entity_type, attributes).and_raise(StandardError)
+      allow(client).to receive(:update!).with(entity_type, primary_key, attributes).and_raise(StandardError)
 
       expect{ subject }.to raise_error(StandardError)
     end
   end
 
   describe '.update!' do
```

Comment:
Can we add a test(s) that evaluates when the `options[:with_metadata] == true`

Reply:
Could you please explain more why should i test it in this place?

Reply:
You are right, there's nothing to test with regards to what I said. My bad :D